stonesoup/moveable/movable.py

```diff
@@ -0,0 +1,532 @@
+        """Return the velocity of the platform.
+
+        For a fixed platform this is always a zero vector of length :attr:`ndim`.
+        """
+        return StateVector([0] * self.ndim)
+
+    @property
+    def is_moving(self) -> bool:
+        return False
+
+    def move(self, timestamp: datetime.datetime, **kwargs) -> None:
+        """For a fixed platform this method has no effect other than to update the timestamp."""
+        # TODO Is this a sensible implementation?
+        # Return without moving static platforms
+        self.state.timestamp = timestamp
+
+
+class MovingMovable(Movable):
+    """Moving platform base class
+
+    A platform represents a random object defined as a :class:`~.State`
+    that moves according to a given :class:`~.TransitionModel`.
+
+    .. note:: Position and orientation are a read only properties in this class.
+    """
+    transition_model: TransitionModel = Property(doc="Transition model")
+
+    @property
+    def velocity(self) -> StateVector:
+        """Return the velocity of the platform.
+
+        Extracted from the state vector of the platform using the platform's
+        :attr:`velocity_mapping`. If the state vector is too short and does not contain the
+        elements specified in the :attr:`velocity_mapping` this raises an :class:`AttributeError`
+        """
+        try:
+            return self.state_vector[self.velocity_mapping, :]
+        except IndexError:
+            raise AttributeError('Velocity is not defined for this platform')
+
+    @property
+    def orientation(self) -> StateVector:
+        """Return the orientation of the platform.
+
+        This is defined as a 3x1 StateVector of angles (rad), specifying the sensor orientation in
+        terms of the counter-clockwise rotation around each Cartesian axis in the order
+        :math:`x,y,z`. The rotation angles are positive if the rotation is in the counter-clockwise
+        direction when viewed by an observer looking along the respective rotation axis,
+        towards the origin.
+
+        The orientation of this platform is defined as along the direction of its velocity, with
+        roll always set to zero (as this is the angle the platform is rotated about the velocity
+        axis, which is not defined in this approximation).
+
+        Notes
+        -----
+        A non-moving platform (``self.is_moving == False``) does not have a defined orientation in
+        this approximations and so raises an :class:`AttributeError`
+        """
+        if not self.is_moving:
+            raise AttributeError('Orientation of a zero-velocity moving platform is not defined')
+        velocity = self.velocity
+
+        if self.ndim == 3:
+            _, bearing, elevation = cart2sphere(*velocity.flat)
+            return StateVector([0, elevation, bearing])
+        elif self.ndim == 2:
+            _, bearing = cart2pol(*velocity.flat)
+            return StateVector([0, 0, bearing])
```

Comment:
For the 2d case should the orientation be just the bearing and the rotation matrix is 2d?

Reply:
The convention throughout is that rotation is handled in 3D. I am unkeen on changing this here, as it's a bit of a big job, and not trivial to get right. I think this is probably out of scope of this PR.